### ISSUE TITLE
[Ideas] Open ideas

### ISSUE THREAD

**Issue by sethtroisi** [tensorflow/minigo#460]
# Ideas inspired by LZ
* SWA: Initial Proof Of Concept in #283 but more work needed
* Visits "timemanagement" (stopping when 2nd move can't overtake first)

# Ideas from AG/AGZ/AZ papers
* Gating: Write up in #459 

# Ideas from elsewhere
* Changing value to 0.95 (AKA 1 - false positive) for selfplay with resign (idea from lightvector)

**Comment by sethtroisi** [comment]
z = z * move_num/length
z = z/2 + q/2
z = z * false_positive_rate in resign disabled games
-----
higher learning rate early

**Comment by sethtroisi** [comment]
Adding stuff about distillation and Seth ideas

**Comment by sethtroisi** [comment]
Checking if eval games have enough diversity and using this opening panel

https://github.com/leela-zero/leela-zero/issues/2104

**Comment by Ishinoshita** [comment]
@sethtroisi Re "timemanagement" from LZ, I'm concerned it might be detrimental for self-play and RL, as it amounts to some sort of policy sharpening: cutting the search early means low policy moves won't get any visit and will be trained towards 0. That may hinder the learning of new stuff.

IMHO, the key to spare compute budget might truly be KataGo's variable visits scheme, for game move search vs policy training target search.

And both types of KataGo's search could benefit from the KLD threshold trick from LC0, that sounds very appealing for policy, though much complex to implement ;-)

**Comment by sethtroisi** [comment]
From Brian Lee:

One concrete idea: instead of selecting 2% flat from the last 50 generations, select 4%->0% over the last 50 generations, with some sort of exponentially decaying curve, and also make this parameter configurable. Early on, we might want to have 10% -> 0% over the last ~10 generations of data, but later on we might want to flatten that curve to select 2% -> 0% over the last 100 generations.
